tools/shared.py

```diff
@@ -236,13 +236,32 @@ def run_js_tool(filename, jsargs=[], node_args=[], **kw):
   return check_call(command, **kw).stdout
 
 
-def get_npm_cmd(name):
+def get_npm_cmd(name_or_path, relative_to_node_bin=True):
+  """Returns a command list that executes the given npm executable.
+
+  If the file does not exist, this is because npm install was not run, so
+  we display that error to the user.
+
+  These executables are typically stored in node_modules/.bin and we
+  return a platform-specific list of a command and args that will invoke it.
+
+  If relative_to_node_bin is True, the first argument will be treated as
+  the name of a node executable in node_modules/.bin. If it is False, the first
+  argument will be used as a path relative to the root directory which is
+  pointing to the node executable.
+  """
```

Comment:
Connecting to the other thread, you clarified that the False case is sent something that is relative to the emscripten install. Is that what you mean by "root directory" in the last sentence? Might be best to write "emscripten root" then maybe.